internal/controllers/machine/machine_controller.go

```diff
@@ -383,6 +401,25 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, s *scope) (ctrl.Result
 	cluster := s.cluster
 	m := s.machine
 
+	// Check reconcileDeleteCache to ensure we won't run reconcileDelete too frequently.
+	if cacheEntry, ok := r.reconcileDeleteCache.Has(cache.NewReconcileEntryKey(s.machine)); ok {
+		if requeueAfter, requeue := cacheEntry.ShouldRequeue(time.Now()); requeue {
+			return ctrl.Result{RequeueAfter: requeueAfter}, nil
+		}
+	}
+
+	s.reconcileDeleteExecuted = true
+
+	// Add entry to the reconcileDeleteCache so we won't run reconcileDelete more than once per second.
+	// Under certain circumstances the ReconcileAfter time will be set to a later time, e.g. when we're waiting
+	// for Pods to terminate or volumes to detach.
+	// This is done to ensure we're not spamming the workload cluster API server.
```

Comment:
If the goal is to protect the workload cluster, I will raise the interval to 3 or 5 seconds.

Reply:
The problem is that this increases the time for a happy path delete by probably 4-5 x (3s or 5s)

We also have client-side rate-limiting